### DISCUSSION TITLE
Change :has() to :-abp-has() as a workaround?

### DISCUSSION THREAD

**Discussion by peace2000** [uBlockOrigin/uAssets#14665] [Ideas]
Due to this bug: https://github.com/uBlockOrigin/uBlock-issues/issues/2228 Wouldn't it be recommended to temporarily change `:has()` to `:-abp-has()` in all nested entries, as a workaround?

**Comment by mapx-** [comment]
@gorhill

**Comment by gorhill** [comment]
I don't think this solve the issue in uBO stable. And as for dev build, we should only change the ones not working as expected to still benefit from the declarative form.

**Comment by gorhill** [comment]
Also, by the way, there is no need to use `##...:-abp-has(...)`, just use `#?#...:has(...)` when the potentially declarative form (which can occur with `##`) does not work.

**Reply by peace2000** [reply]
How long it'll take for the majority of uBO users to update uBO to 1.44.2 ? I'm worrying about those who still use the older versions. For them `#?#...:has(...)` doesn't guarantee that the filter will be treated as a procedural.

**Reply by krystian3w** [reply]
> about those who still use the older versions. 

Then recommended is update add-on instead stay on outdated add-on without important reason.

> users to update uBO to 1.44.2

Maybe my private blocklist can (like pizza90 or eladkarako) disable paste found bugs (IMO rare chance they group testing vanilla `:has` vs JavaScript `:has` and waiting when issue unlock commenting) without duplicate or asking friends to repaste.

**Reply by peace2000** [reply]
> > about those who still use the older versions.
> 
> Then recommended is update add-on instead stay on outdated add-on without important reason.
> 
True, but I'm not speaking about those who willingly stay in outdated versions, it takes a while for the updates to reach everyone.

**Reply by krystian3w** [reply]
Then needed is documented the biggest breakage - IMO no exist bigger than back sponsored posts on Instagram due Google fail (Meta don't implemented built word similar like on Facebook or eBay).
